src/Core/Grid/Query/OrderQueryBuilder.php

```diff
@@ -0,0 +1,274 @@
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Query;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PDO;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+
+/**
+ * Builds SQL queries using Doctrine for retrieving data for orders grid
+ */
+final class OrderQueryBuilder implements DoctrineQueryBuilderInterface
+{
+    /**
+     * @var Connection
+     */
+    protected $connection;
+
+    /**
+     * @var string
+     */
+    protected $dbPrefix;
+
+    /**
+     * @var int
+     */
+    private $contextLangId;
+
+    /**
+     * @var DoctrineSearchCriteriaApplicatorInterface
+     */
+    private $criteriaApplicator;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param DoctrineSearchCriteriaApplicatorInterface $criteriaApplicator
+     * @param int $contextLangId
+     */
+    public function __construct(
+        Connection $connection,
+        $dbPrefix,
+        DoctrineSearchCriteriaApplicatorInterface $criteriaApplicator,
+        $contextLangId
+    ) {
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+        $this->contextLangId = $contextLangId;
+        $this->criteriaApplicator = $criteriaApplicator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $newCustomerSubSelect = $this->connection
+            ->createQueryBuilder()
+            ->select('so.id_order')
+            ->from($this->dbPrefix . 'orders', 'so')
+            ->where('so.id_customer = o.id_customer')
+            ->andWhere('so.id_order < o.id_order')
+            ->setMaxResults(1)
+        ;
+
+        $qb = $this
+            ->getBaseQueryBuilder($searchCriteria->getFilters())
+            ->addSelect('o.id_order, o.reference, o.total_paid_tax_incl, os.paid, osl.name AS osname')
+            ->addSelect('CONCAT(LEFT(cu.`firstname`, 1), \'. \', cu.`lastname`) AS `customer`')
+            ->addSelect('os.color, o.payment')
+            ->addSelect('o.date_add, cu.company, cl.name AS country_name, o.invoice_number, o.delivery_number')
+            ->addSelect('IF ((' . $newCustomerSubSelect->getSQL() . ') > 0, 0, 1) AS new')
```

Comment:
We'll have to check that this statement does not introduce performance issues. Subqueries can be terrible memory and time consumers